src/main/java/ch/uzh/ifi/hase/soprafs22/screwyourneighborserver/sideeffects/HandEventHandler.java

```diff
@@ -26,9 +28,11 @@ public HandEventHandler(MatchRepository matchRepository) {
   @HandleBeforeSave
   public void onBeforeSave(Hand hand) {
     Match match = hand.getMatch();
-    boolean scoreAllowed = (match.getSumOfScoreAnnouncement() - hand.getCards().size()) == 0;
-    if (match.isLastAnnouncement() && scoreAllowed) {
+    boolean scoreAllowed = (match.getSumOfScoreAnnouncement() - hand.getCards().size()) != 0;
+    if (match.isLastAnnouncement() && !scoreAllowed) {
       hand.setAnnouncedScore(empty);
+      throw new HttpClientErrorException(
+          HttpStatus.FORBIDDEN, "Game rules prohibit this score announcement");
```

Comment:
```suggestion
          HttpStatus.UNPROCESSABLE_ENTITY, "Game rules prohibit this score announcement");
```
Http status codes:
400: BAD_REQUEST (z.b. String statt integer für eine property, property fehlt)
401: UNAUTHORIZED: Keine Session, somit konnte der Zugriff nicht authorisiert werden
403: FORBIDDEN: Session vorhanden, aber mit dieser authorisierung kommst du hier nicht rein. D.h der gleiche Request mit einer anderen Berechtigung müsste funktionieren.
404: NOT FOUND: Was du suchst gibt es nicht
422: UNPROCESSABLE_ENTITY: Alle properties da mit dem richtigen typ, aber der Request kann nicht verarbeitet werden, weil er nicht zum momentanen Stand der DB passt
(Man kann streiten ob 409 dafür richtig ist. Ich habe leider schon mit 422 angefangen, dann würde ich es konsistent halten)

https://en.wikipedia.org/wiki/List_of_HTTP_status_codes

Reply:
Die hab' I gsuächt! 👍

Reply:
Wo wirfst du die 422 schon wieder?